### DISCUSSION TITLE
How to switch Spaces in Bunch?

### DISCUSSION THREAD

**Discussion by VladGhitulescu** [ttscoff/bunch-docs#203] [Q&A] [answered]
Hey!

My second Bunch 

1. opens a specific EagleFiler-library then…
2. … waits until this is done then…
3. … opens MailMate,
4. … opens a specific folder in a Finder-window, and after this is also done then…
5. … position the Finder-window and the MailMate-window according to a specific Moom-windows-layout.6. 

I would like to start the Bunch in a Space adjacent to the one where MailMate is already open, do the first two steps (= open the EagleFiler-library) and after this switch to the adjacent Space where MailMate is eagerly waiting ;-) for the next 3 steps, and only then execute them too.

So what I want is to switch to the adjacent Space (CTRL - left arrow) between steps 2 and 3 above (see the red arrow in the screenshot below):

https://www.dropbox.com/s/a2mjri2egygo53i/Bildschirmfoto%202022-02-27%20um%2015.01.49.png?dl=0

How can I do this in Bunch?

I don't know how to set keyboard-shortcuts in Bunch yet.

I could create a KeyboardMaestro-Macro that only does the Spaces-switch… but I don't know how to use a KeyboardMaestro-Macro in Bunch either yet! :-(

Or is there perhaps a much better way to do this?!

I attach the Bunch at the bottom.

Thanks!

Regards,
Vlad

---

```
---
title: Archiviere E-Mails
---


EagleFiler

%Finder
- /Volumes/Data-1/EagleFiler-Bibliotheken/MailMate/MailMate.eflibrary

    <<#delayedMailMate&Finder

    <<#DelayMoom


___



---[delayedMailMate&Finder]
%MailMate

%Finder
- /Volumes/Data-1/Backup/Mail Archive/MailMate-Archive

---[DelayMoom]
* tell application "Moom" to arrange windows according to snapshot "MailMateArchiv"
```

**Comment by ttscoff** [comment, marked as answer]
You can send keyboard shortcuts just by using:

{ctrl-right}

In the Bunch. You can also easily trigger km macros using their url handler. Just put the url on its own line in the Bunch.

The problem you’ll probably run into, though, is sequencing. Even if the Bunch is set to sequential execution order, the speed at which the system and various apps respond to commands can make them appear to happen out of order. This can only be mitigated using delays and pause commands, which can take some fiddling.

- Brett

**Comment by VladGhitulescu** [comment]
Hey!


Thanks for the quick reply!

I must somehow missed the fact that I can send keyboard shortcuts 
through Bunch itself.
I tried it and got this error-Message

***@***.*** 
"Bildschirmfoto 2022-02-27 um 21.24.10.png")


I forgot to say that I'm using Bunch on a late 2012 iMac running macOS 
Mojave Version 10.14.6.

And yes, I also got some problems with the delays (after starting 
EagleFiler)

I think I'll ditch the idea to switch Spaces until Apple brings the new 
iMac 27" / Pro I'm sorely missing and I'll run this Bunch on a single 
Space until then.

OT: It would be great if you could find a little time to write about the 
current configuration of your Ultimate Hacking Keyboard - I received 
mine a week or so ago (after waiting since April last year!) and I'm 
programming as much keys as I can ;-)

Thanks again!


Regards,
Vlad

**Comment by ttscoff** [comment]
Yeah, now that the UHK is actually shipping new decks (with modules!) out at a decent rate, I should write an updated article. You're not the first to ask :)

**Comment by VladGhitulescu** [comment]
Yes, that would improve everybody's settings for the UHK since there is no place to take a look at smart configurations or best practices. 

I can imagine there's also a need for something like a forum - László said something about Discourse planned for later.  

I'm using my UHK with my iMac and my work PC and I'm trying to keep the two configurations as similar as possible. 

Thanks again, I can't wait!



> Am 08.03.2022 um 18:32 schrieb Brett Terpstra ***@***.***>:
> 
> ﻿
> Yeah, now that the UHK is actually shipping new decks (with modules!) out at a decent rate, I should write an updated article. You're not the first to ask :)
> 
> On 27 Feb 2022, at 14:37, VladGhitulescu wrote:
> 
> > Hey!
> >
> >
> > Thanks for the quick reply!
> >
> > I must somehow missed the fact that I can send keyboard shortcuts
> > through Bunch itself.
> > I tried it and got this error-Message
> >
> > ***@***.***
> > "Bildschirmfoto 2022-02-27 um 21.24.10.png")
> >
> >
> > I forgot to say that I'm using Bunch on a late 2012 iMac running macOS
> > Mojave Version 10.14.6.
> >
> > And yes, I also got some problems with the delays (after starting
> > EagleFiler)
> >
> > I think I'll ditch the idea to switch Spaces until Apple brings the new
> > iMac 27" / Pro I'm sorely missing and I'll run this Bunch on a single
> > Space until then.
> >
> > OT: It would be great if you could find a little time to write about the
> > current configuration of your Ultimate Hacking Keyboard - I received
> > mine a week or so ago (after waiting since April last year!) and I'm
> > programming as much keys as I can ;-)
> >
> > Thanks again!
> >
> >
> > Regards,
> > Vlad
> >
> >
> >
> >
> >
> >
> >
> > On 27 Feb 2022, at 18:45, Brett Terpstra wrote:
> >
> >> You can send keyboard shortcuts just by using:
> >>
> >> {ctrl-right}
> >>
> >> In the Bunch. You can also easily trigger km macros using their url
> >> handler. Just put the url on its own line in the Bunch.
> >>
> >> The problem you’ll probably run into, though, is sequencing. Even if
> >> the Bunch is set to sequential execution order, the speed at which the
> >> system and various apps respond to commands can make them appear to
> >> happen out of order. This can only be mitigated using delays and pause
> >> commands, which can take some fiddling.
> >>
> >> - Brett
